### ISSUE TITLE
Safe mode/encourage making schema changes on the lowest environment on the Cloud

### ISSUE THREAD

**Issue by Jay-umbr** [umbraco/Umbraco.Cloud.Issues#248]
With the way the Cloud is set up - where schema between environments is synchronized by Git deployments - it is possible to mess up your project quite a bit by making changes directly on Live, and then deploying from Dev to Live (if you happen to have more than one environment). It would be nice if the users would have the option to enable/disable a "safe mode" - and in that case, let's say:

-you have Dev and Live environments
-you have safe mode enabled
-you log into Dev backoffice, can make changes in doctypes/templates/datatypes etc. just fine
-you log into Live backoffice, but whenever you try to make a change/save changes in schema on Live, a popup would appear saying something like

"Safe mode is enabled. You cannot save any schema changes to prevent possible conflicts between environments. If you wish to modify the schema, please make the changes on the Development Environment and deploy them all the way up.
To disable safe mode please go to" ... etc etc

With this setting in place, and enabled, it would enforce the good practice of making changes on Dev instead of Live, if you have more than one environment. That could prevent schema mismatch errors, and colliding datatypes/templates/etc,

**Comment by nul800sebastiaan** [comment]
Interesting, that would probably require us to significantly change how Umbraco works. 🙈

**Comment by sitereactor** [comment]
It's something we've talked about before in relation to vNext and having the Live environment somewhere else. If the Settings section could be read-only it would better support the flow of changes (typically done by a developer in the dev-environment or local) are pushed forward without the risk of something having been changed in another environment.

While we can remove the sections to enforce this it would be a lot nicer if certain sections could be set to read-only.

**Comment by nul800sebastiaan** [comment]
This will go on our long-term ideas list then for now! 👍

**Comment by skttl** [comment]
That's a nice idea. I think it can be done with event handlers on the different save events if you need it now :)

**Comment by thomasknielsen** [comment]
In the same lines - it might be worth considering to make it possible to disable the automatic patch updates of Umbraco if you have a local dev environment often it would be nice to control what version of Umbraco is used and plan the upgrade to push out from local->dev->live.

**Comment by sitereactor** [comment]
@skttl yes it could be done by cancelling events, but I would prefer a friendly'er approach where its obvious that what you are looking at is read-only. Maybe greyed out icons and editor, maybe a banner of sorts and no save button.

@thomasknielsen I think that is a bit of a stretch 😉 but scheduling updates is something that is in the cards. It's still early though, so difficult to say how it would work, but I imagine some kind of schedule per Project, which the Project Owner can control, so updates are either applied right away or on a controlled schedule.

**Comment by robmaas** [comment]
I have described a similar FR on Umbraco-CMS. It's not a Cloud-only thing, however the implementation will probably be a bit different for Cloud projects.
https://github.com/umbraco/Umbraco-CMS/issues/8862

**Comment by matthewcare** [comment]
This is something I'd like to see also.
My current approach for restricting access is with a rewrite rule such as
```
<rule name="Block Umbraco Edit Screens">
  <match url="*" />
  <conditions logicalGrouping="MatchAny">
    <add input="{URL}" pattern="/umbraco/views/documentTypes/edit.html" />
     .....
  </conditions>
  <action type="Rewrite" url="/restricted.html"  />
</rule>
```
so it would be nice to have a proper way to disable it.
